nomad/variables_endpoint_test.go

```diff
@@ -22,12 +22,278 @@ import (
+
+		must.NoError(t, err)
+		must.Eq(t, structs.VarOpResultRedacted, applyResp.Result)
+		must.Eq(t, svHold.VariableMetadata, applyResp.Conflict.VariableMetadata)
+		must.Nil(t, applyResp.Conflict.Items)
+	})
+
+	t.Run("cas/write only/read own upsert", func(t *testing.T) {
+		must.NotNil(t, svHold)
+		sv := svHold
+		sv.Items["upsert"] = "read"
+
+		applyReq := structs.VariablesApplyRequest{
+			Op:  structs.VarOpCAS,
+			Var: sv,
+			WriteRequest: structs.WriteRequest{
+				Region:    "global",
+				AuthToken: writeToken.SecretID,
+			},
+		}
+		applyResp := new(structs.VariablesApplyResponse)
+		err := msgpackrpc.CallWithCodec(codec, structs.VariablesApplyRPCMethod, &applyReq, &applyResp)
+
+		must.NoError(t, err)
+		must.Eq(t, structs.VarOpResultOk, applyResp.Result)
+		must.Eq(t, sv.Items, applyResp.Output.Items)
+	})
+
+	t.Run("lock-acquire/management token/new", func(t *testing.T) {
+		must.NotNil(t, svHold)
+		sv := *svHold
+
+		sv.Items["upsert"] = "read"
+		sv.VariableMetadata.Lock = &structs.VariableLock{
+			TTL:       24 * time.Hour,
+			LockDelay: 15 * time.Second,
+		}
+
+		applyReq := structs.VariablesApplyRequest{
+			Op:  structs.VarOpLockAcquire,
+			Var: &sv,
+			WriteRequest: structs.WriteRequest{
+				Region:    "global",
+				AuthToken: rootToken.SecretID,
+			},
+		}
+
+		applyResp := new(structs.VariablesApplyResponse)
+		err := msgpackrpc.CallWithCodec(codec, structs.VariablesApplyRPCMethod, &applyReq, &applyResp)
+
+		must.NoError(t, err)
+		must.Eq(t, structs.VarOpResultOk, applyResp.Result)
+		must.NonZero(t, len(applyResp.Output.VariableMetadata.Lock.ID))
+
+		svHold = applyResp.Output
+	})
+
+	t.Run("lock-release/management token/new", func(t *testing.T) {
+		must.NotNil(t, svHold)
+		sv := svHold
+
+		applyReq := structs.VariablesApplyRequest{
+			Op:  structs.VarOpLockRelease,
+			Var: sv,
+			WriteRequest: structs.WriteRequest{
+				Region:    "global",
+				AuthToken: rootToken.SecretID,
+			},
+		}
+
+		applyResp := new(structs.VariablesApplyResponse)
+		err := msgpackrpc.CallWithCodec(codec, structs.VariablesApplyRPCMethod, &applyReq, &applyResp)
+
+		must.NoError(t, err)
+		must.Eq(t, structs.VarOpResultOk, applyResp.Result)
+		must.Nil(t, applyResp.Output.VariableMetadata.Lock)
+	})
+
+	t.Run("lock-acquire/read token/new", func(t *testing.T) {
+		must.NotNil(t, svHold)
+		sv := *sv3
+
+		sv.Items["upsert"] = "read"
+		sv.VariableMetadata.Lock = &structs.VariableLock{
+			TTL:       24 * time.Hour,
+			LockDelay: 15 * time.Second,
+		}
+
+		applyReq := structs.VariablesApplyRequest{
+			Op:  structs.VarOpLockAcquire,
+			Var: &sv,
+			WriteRequest: structs.WriteRequest{
+				Region:    "global",
+				AuthToken: readToken.SecretID,
+			},
+		}
+
+		applyResp := new(structs.VariablesApplyResponse)
+		err := msgpackrpc.CallWithCodec(codec, structs.VariablesApplyRPCMethod, &applyReq, &applyResp)
+		must.EqError(t, err, structs.ErrPermissionDenied.Error())
+	})
```

Comment:
If the lock acquire was successful, it will always have the lock ID, and no matter if the release was successful or not, it will never return the lock ID, no matter if the token is management or not. The show or no show is more important for the list and get endpoints and this test is for the apply one.